zaza/openstack/charm_tests/gnocchi/tests.py

```diff
@@ -110,3 +113,38 @@ def test_s3_list_gnocchi_buckets(self):
                     break
                 else:
                     AssertionError('Bucket "{}" not found'.format(gnocchi_bkt))
+
+
+class GnocchiExternalCATest(test_utils.OpenStackBaseTest):
+    """Test Gnocchi for external root CA config option."""
+
+    def test_upload_external_cert(self):
+        """Verify that the external CA is uploaded correctly."""
+        logging.info('Changing value for trusted-external-ca-cert.')
+        ca_cert_option = 'trusted-external-ca-cert'
+        ppk, cert = utilities.cert.generate_cert('gnocchi_test.ci.local')
+        b64_cert = base64.b64encode(cert).decode()
+        config = {
+            ca_cert_option: b64_cert,
+        }
+        model.set_application_config(
+            'gnocchi',
+            config
+        )
+        model.block_until_all_units_idle()
+
+        cert_location = '/usr/local/share/ca-certificates'
+        cert_name = 'gnocchi-external.crt'
+        cmd = 'ls ' + cert_location + '/' + cert_name
```

Comment:
Let's use block_until_file_has_contents or block_until_file_ready rather than a simple ls.

https://github.com/openstack-charmers/zaza/blob/master/zaza/model.py#L1246
https://github.com/openstack-charmers/zaza/blob/master/zaza/model.py#L1288

Reply:
I updated the PR using block_until_file_has_contents.